streams/src/main/java/org/apache/kafka/streams/processor/internals/StateDirectory.java

```diff
@@ -288,50 +277,31 @@ private String logPrefix() {
      * Get the lock for the {@link TaskId}s directory if it is available
      * @param taskId task id
      * @return true if successful
-     * @throws IOException if the file cannot be created or file handle cannot be grabbed, should be considered as fatal
      */
-    synchronized boolean lock(final TaskId taskId) throws IOException {
+    synchronized boolean lock(final TaskId taskId) {
         if (!hasPersistentStores) {
             return true;
         }
 
-        final File lockFile;
-        // we already have the lock so bail out here
-        final LockAndOwner lockAndOwner = locks.get(taskId);
-        if (lockAndOwner != null && lockAndOwner.owningThread.equals(Thread.currentThread().getName())) {
-            log.trace("{} Found cached state dir lock for task {}", logPrefix(), taskId);
+        final String lockOwner = lockedTasksToStreamThreadOwner.get(taskId);
+        if (lockOwner != null) {
+            if (lockOwner.equals(Thread.currentThread().getName())) {
```

Comment:
seems dangerous to use the thread name to check ownership. might be safer to check reference equality on `Thread`

Reply:
never mind - I see this isn't a change from what was happening before. Still seems good to do, but not necessarily in this patch

Reply:
Yeah I agree, I filed a ticket to improve the locking mechanism a little while back. Just wanted to keep this PR focused

Reply:
Hm I can't seem to find that ticket, maybe I just thought about filing one. Anyways here's https://issues.apache.org/jira/browse/KAFKA-12509